src/decoder.ts

```diff
@@ -383,6 +382,72 @@ export class Decoder<A> {
     });
   }
 
+  /**
+   * Decoder for fixed-length arrays, aka Tuples.
+   *
+   * Supports up to 8-tuples.
+   *
+   * Example:
+   * ```
+   * tuple([number(), number(), string()]).run([5, 10, 'px'])
+   * // => {ok: true, result: [5, 10, 'px']}
+   * ```
+   */
+  static tuple<A>(decoder: [Decoder<A>]): Decoder<[A]>;
+  static tuple<A, B>(decoder: [Decoder<A>, Decoder<B>]): Decoder<[A, B]>;
```

Comment:
Ok, so this is very interesting to me. Here's what I was imagining when I first suggested we add this decoder:
```
static tuple = <A>(decoders: DeciderObject<A>): Decoder<A> => ...
``` 
This method uses the `DecoderObject` type to map over a tuple of type `A`, resulting in a tuple where each element is a decoder for the element type. That's great because we can handle tuples of any length.

That said, there are reasons why we might want to stick with your implementation. First off, when do you really need more than an 8-tuple? Secondly, using a mapped type in this way does have some downsides. Something I've ran into is that Typescript is much better at forward type inference than backwards type inference. The distinction looks like this:
```
const forwardInference = <T>(arg: T): SomeTypeOperation<T> => ...

const backwardsInference = <T>(arg: SomeTypeOperation<T>): T => ...
```
because the `object` decoder using backwards inference, there are some edge case situations where TS will fail to infer the decoder type, and give a rally bad error message if you don't annotate the decoder like `object<MyObjectType>({...})`.

I'm not yet sure if there are major downsides to having explicit tuple types, like you've done here. I'll think about it for a bit. What do you think?